Make the 3.1 notes more prominent in kubernetes-events.md as this is a breaking change

douglasawh opened the pull request:
This is a pretty massive change in the way the plugin works. Sticking this at the bottom in a foot note does not seem appropriate.

The decision to do this initially does make some sense, as 3.1 is not going to be the version that everyone is using. However, over time, this will come to be the dominant behavior, and that behavior should be made clear to the users.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Documentation**
  * Updated Kubernetes events configuration documentation with improved clarity
  * Refined configuration parameters table including updates to default values for namespace, TLS, and synchronization settings
  * Enhanced descriptions for better understanding of available configuration options

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/fluent/fluent-bit-docs/pull/1449?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

douglasawh committed:
Update kubernetes-events.md

This is a pretty massive change in the way the plugin works. Sticking this at the bottom in a foot note does not seem appropriate.

The decision to do this initially does make some sense, as 3.1 is not going to be the version that everyone is using. However, over time, this will come to be the dominant behavior, and that behavior should be made clear to the users.

Signed-off-by: Doug Whitfield <dwhitfield@perforce.com>

lockewritesdocs commented on pipeline/inputs/kubernetes-events.md:
```diff
@@ -9,6 +9,7 @@ Kubernetes exports it events through the API server. This input plugin allows to
 
 ## Configuration
 
+NOTE: As of Fluent-Bit 3.1, this plugin uses a Kubernetes watch stream instead of polling. In versions before 3.1, the interval parameters are used for reconnecting the Kubernetes watch stream.
```
```suggestion
Note: Starting with Fluent Bit 3.1, the Kubernetes Events plugin uses a Kubernetes
watch stream instead of polling. Previous versions use the interval parameters for
reconnecting the Kubernetes watch stream.
```

lockewritesdocs reviewed (changes requested):
Thanks for your contribution @douglasawh! I suggested some editorial changes to incorporate.

douglasawh committed:
Update pipeline/inputs/kubernetes-events.md

This language much better than my original PR. All I *had* done is take what was in there, reformat it slightly, and move it to the top.

Co-authored-by: Adam Locke <adam.locke@chronosphere.io>
Signed-off-by: Doug Whitfield <douglasawh@gmail.com>

lecaros commented:
Hi @douglasawh,
thanks for your submission.
Could you please review our [contributing guidelines](https://github.com/fluent/fluent-bit-docs/blob/master/CONTRIBUTING.md) and adhere to them? After that, we'll gladly merge your contribution.

Thanks

esmerel committed:
Merge branch 'master' into patch-1

Signed-off-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>

esmerel commented on pipeline/inputs/kubernetes-events.md:
```diff
@@ -8,6 +8,9 @@ Kubernetes exports events through the API server. This input plugin lets you ret
 
 ## Configuration
 
+Note: Starting with Fluent Bit 3.1, the Kubernetes Events plugin uses a Kubernetes
```
```suggestion
Starting with Fluent Bit 3.1, the Kubernetes Events plugin uses a Kubernetes
```

esmerel committed:
Update pipeline/inputs/kubernetes-events.md

Signed-off-by: Lynette  Miles <6818907+esmerel@users.noreply.github.com>

eschabell committed:
Merge branch 'master' into patch-1

Signed-off-by: Eric D. Schabell <eric@schabell.org>
